ibc-core/ics04-channel/src/handler/acknowledgement.rs

```diff
@@ -1,19 +1,95 @@
+use ibc_core_channel_types::acknowledgement::Acknowledgement;
 use ibc_core_channel_types::channel::{Counterparty, Order, State as ChannelState};
 use ibc_core_channel_types::commitment::{compute_ack_commitment, compute_packet_commitment};
 use ibc_core_channel_types::error::ChannelError;
-use ibc_core_channel_types::events::AcknowledgePacket;
+use ibc_core_channel_types::events::{AcknowledgePacket, WriteAcknowledgement};
 use ibc_core_channel_types::msgs::MsgAcknowledgement;
+use ibc_core_channel_types::packet::{Packet, Receipt};
 use ibc_core_client::context::prelude::*;
 use ibc_core_connection::delay::verify_conn_delay_passed;
 use ibc_core_connection::types::State as ConnectionState;
 use ibc_core_handler_types::events::{IbcEvent, MessageEvent};
 use ibc_core_host::types::path::{
-    AckPath, ChannelEndPath, ClientConsensusStatePath, CommitmentPath, Path, SeqAckPath,
+    AckPath, ChannelEndPath, ClientConsensusStatePath, CommitmentPath, Path, ReceiptPath,
+    SeqAckPath, SeqRecvPath,
 };
 use ibc_core_host::{ExecutionContext, ValidationContext};
 use ibc_core_router::module::Module;
 use ibc_primitives::prelude::*;
 
+pub fn commit_packet_sequence_no<ExecCtx>(
```

Comment:
Maybe a better naming for this function? what's the `*_no`?

Reply:
`*_no` was in reference to "number". I'll improve the fn name here.